WindowsStubs/WindowsStubs/TaskUtil.cpp

```diff
@@ -0,0 +1,107 @@
+#include <windows.h>
+#include <stdio.h>
+#include <tchar.h>
+#include <iostream>
+#include<string>
+#include <cctype>
+#include <algorithm>
+
+std::string TaskExecuteAndWait(std::string execpath, bool usePipes)
+{
+	
+	STARTUPINFO si;
+	PROCESS_INFORMATION pi;
+	SECURITY_ATTRIBUTES securityAttr;
+	HANDLE inputPipe, outputPipe;
+	std::string  output;
+
+	ZeroMemory(&si, sizeof(si));
+	si.cb = sizeof(si);
+	ZeroMemory(&pi, sizeof(pi));
+	ZeroMemory(&securityAttr, sizeof(securityAttr));
+	securityAttr.nLength = sizeof(securityAttr);
+	securityAttr.bInheritHandle = TRUE;
+
+
+
+	//std::wstring wexecpath;
+	//std::copy(str.begin(), str.end(), std::back_inserter(wexecpath));
+
+	int len;
+	int slength = (int)execpath.length() + 1;
+	len = MultiByteToWideChar(CP_ACP, 0, execpath.c_str(), slength, 0, 0);
+	wchar_t* buf = new wchar_t[len];
+	MultiByteToWideChar(CP_ACP, 0, execpath.c_str(), slength, buf, len);
+	//std::wstring r(buf);
+	//delete[] buf;
+	//return r;
+	
+
+	if (usePipes)
+	{
+
+
+		::CreatePipe(&inputPipe, &outputPipe, &securityAttr, 0);
+
+		si.dwFlags = STARTF_USESTDHANDLES;
+		si.hStdInput = NULL;
+		si.hStdOutput = outputPipe;
+		si.hStdError = outputPipe;
+	}
+
+	if (!CreateProcess(NULL,   // No module name (use command line)
+		buf,        // Command line
+		NULL,           // Process handle not inheritable
+		NULL,           // Thread handle not inheritable
+		TRUE,          // Set handle inheritance to FALSE
+		0,              // No creation flags
+		NULL,           // Use parent's environment block
+		NULL,           // Use parent's starting directory 
+		&si,            // Pointer to STARTUPINFO structure
+		&pi)           // Pointer to PROCESS_INFORMATION structure
+		)
```

Comment:
Will make it difficult to read. And difficult to understand parameters if comments are removed.